devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -555,13 +559,40 @@ protected void visitProjectDependencies(Project project, Project dep, Set<String
             }
             compilePlusRuntimeConfig.getIncoming().getDependencies()
                     .forEach(d -> {
+                        Project depProject = null;
+
                         if (d instanceof ProjectDependency) {
-                            visitProjectDep(project, ((ProjectDependency) d).getDependencyProject(), visited);
+                            depProject = ((ProjectDependency) d).getDependencyProject();
+                        } else if (d instanceof ExternalModuleDependency) {
+                            depProject = ToolingUtils.findIncludedProject(project, (ExternalModuleDependency) d);
+                        }
+
+                        if (depProject == null) {
+                            return;
+                        }
+
+                        if (depProject.getState().getExecuted()) {
+                            visitPossibleExtensionProject(project, depProject, visited);
+                        } else {
+                            depProject.afterEvaluate(p -> visitPossibleExtensionProject(project, p, visited));
```

Comment:
Isn't this change limiting dependency processing to local extension subprojects compared to the current impl?

Reply:
This is called as part of the process of determining a local module's build dependencies, so that they also get built before building the module.

So if I have a module using the application plugin and want to use a local extension, I want to make sure that the local extension and its local module dependencies are also built before building the application.

This doesn't affect how external extensions (e.g. from Maven) are used in any way.

Reply:
Right, my question was about visiting non-extension local dependencies.

Reply:
It looks like `visitProjectDep` is currently called for any local dependency, while with this change it will be called only for local extension dependencies?

Reply:
Yeah, that makes sense now, thanks.